### PR TITLE
Enabling Istanbul hardfork

### PR DESCRIPTION
**Pull request by mrsmkl**
### Description

Enables Istanbul hard fork, except some gas changes.

### CODE REVIEW

File: packages/celotool/src/e2e-tests/transfer_tests.ts [hunk 1]
```diff
@@ -204,7 +205,7 @@ describe('Transfer tests', function(this: any) {
   const ToAddress = '0xbBae99F0E1EE565404465638d40827b54D343638'
   const FeeRecipientAddress = '0x4f5f8a3f45d179553e7b95119ce296010f50f6f1'
 
-  const syncModes = ['full', 'fast', 'light', 'lightest']
+  const syncModes = ['full' /*, 'fast', 'light', 'lightest' */]
```
Comment (nategraf) [inline packages/celotool/src/e2e-tests/transfer_tests.ts]: We need to revert this change before merging.
```suggestion
  const syncModes = ['full', 'fast', 'light', 'lightest']
```

File: packages/protocol/package.json [hunk 2]
```diff
@@ -40,7 +40,7 @@
     "@0x/sol-profiler": "^3.0.0",
     "@0x/sol-trace": "^2.0.16",
     "@0x/subproviders": "^5.0.0",
-    "@celo/ganache-cli": "git+https://github.com/celo-org/ganache-cli.git#d5ff1f1",
+    "@celo/ganache-cli": "git+https://github.com/mrsmkl/ganache-cli.git#630cf0ee6249b872c209b8da5be2f7f24a959258",
```
Comment (nategraf) [inline packages/protocol/package.json]: Do we also need a PR for our fork of `ganache`?

**Comment by nategraf** [comment]
Nice work. I've left a couple comments.

**Review by asaj — APPROVED** [review]
